### DISCUSSION TITLE
How to verify a Date.toISOString() with MatchV3.datetime

### DISCUSSION THREAD

**Discussion by long-bai** [pact-foundation/pact-js#1204] [Q&A]
Hello everyone,

I have a problem on my consumer to verify the date on ISO string. suppose that i have an attribute date with the value like '2015-08-06T16:53:10Z', and on my consumer i defined with MatchersV3.datetime("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'", '1988-03-12T12:08:10.000Z')
so when i try to verify with my provider, i have an error like `Expected '2023-10-02T23:00:00.000Z' to match a timestamp pattern of 'yyyy-MM-dd'T'HH:mm:ss.SSSZ': Error(Nom("Z", IsA))`

Everyone has the solution for this? Thanks

**Comment by portyu9** [comment]
Hi @long-bai 

For a value produced by:

```js
new Date().toISOString()
```

use the ISO-8601 timezone token `X` instead of quoting `Z` as a literal.

With current Pact JS:

```js
const { Matchers } = require('@pact-foundation/pact')

const { timestamp } = Matchers

const body = {
    date: timestamp(
        "yyyy-MM-dd'T'HH:mm:ss.SSSX",
        '1988-03-12T12:08:10.000Z'
    )
}
```

The important part is:

```text
yyyy-MM-dd'T'HH:mm:ss.SSSX
```

`X` represents an ISO-8601 timezone. For UTC, it accepts:

```text
Z
```

So a provider value such as:

```text
2023-10-02T23:00:00.000Z
```

matches that format.

Your original pattern was:

```js
"yyyy-MM-dd'T'HH:mm:ss.SSS'Z'"
```

which treats `Z` as literal text rather than as the timezone component. That is the wrong representation for an ISO timestamp matcher and is what leads to the format-matching problem you're seeing.

The current Pact JS docs use this exact form:

```js
timestamp("yyyy-MM-dd'T'HH:mm:ss.SSSX")
```

You can also use the ISO-8601 convenience matcher when you do not need a custom format:

```js
const { iso8601DateTimeWithMillis } = Matchers

const body = {
    date: iso8601DateTimeWithMillis(
        '1988-03-12T12:08:10.000Z'
    )
}
```

So for `Date.toISOString()`, prefer `...SSSX` rather than a quoted `'Z'`.

Please mark this answer as accepted if it helped, thank you.
